viennacl/src/main/scala/org/apache/mahout/viennacl/opencl/javacpp/DenseColumnMatrix.scala

```diff
@@ -36,12 +36,24 @@ import org.bytedeco.javacpp.annotation._
 @Name(Array("viennacl::matrix<double,viennacl::column_major>"))
 final class DenseColumnMatrix(initDefault:Boolean = true) extends MatrixBase {
 
-  def this(nrow: Int, ncol: Int, ctx: Context = new Context()) {
+  def this(nrow: Int, ncol: Int) {
+    this(false)
+    allocate(nrow, ncol, new Context())
+  }
+
+  def this(nrow: Int, ncol: Int, ctx: Context) {
     this(false)
     allocate(nrow, ncol, ctx)
   }
 
-  def this(data: DoublePointer, nrow: Int, ncol: Int, ctx: Context = new Context(Context.MAIN_MEMORY)) {
```

Comment:
rebuilding abd testing now with the correct scala flags.. noticed this in spark/dependency-reduced-pom.xmlpom.xml:  [INFO] Reading assembly descriptor: src/main/assembly/dependency-reduced.xml

```
[WARNING] The following patterns were never triggered in this artifact inclusion filter:
o  'org.apache.mahout:mahout-native-viennacl_2.10'
o  'org.apache.mahout:mahout-native-viennacl-omp_2.10'

[INFO] Building jar: /home/andy/sandbox/mahout_cuda/spark/target/mahout-spark_2.11-0.13.1-SNAPSHOT-dependency-reduced.jar
```
*easyhack*: add
to `spark/dependency reduced.xml`:
```
  'org.apache.mahout:mahout-native-viennacl_2.11'
  'org.apache.mahout:mahout-native-viennacl-omp_2.11'
```
though that pom should honor the `{scala.compat.version}`